File: evento/src/main/java/com/myevent/eventos/entity/Vendedor.java
package com.myevent.eventos.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Vendedor {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	private String nome;

	@ManyToOne
	private Departamento departamento;

	private String role;

	private Boolean status;


	private String usuario;

	private String password;

}

File: evento-app/evento-app/src/app/sistema/vendedores/Vendedor.ts
import { Departamento } from "../departamento/Departamento";

export class Vendedor {
    id: number;
    nome: string;
    role: string;
    departamentoId: number
    usuario: string;
    password: string;
    status: boolean;

    constructor(id: number,
        nome: string,
        role: string,
        departamentoId: number,
        usuario: string,
        password: string,
        status: boolean
)    {
        this.id = id;
        this.nome = nome;
        this.role = role;
        this.departamentoId = departamentoId;
        this.usuario = usuario;
        this.password = password;
        this.status = status;

    }
}

export class VendedorDTO {
    id: number;
    nome: string;
    role: string;
    departamento: Departamento
    usuario: string;
    password: string;
    status: boolean;

    constructor(id: number,
        nome: string,
        role: string,
        departamento: Departamento,
        usuario: string,
        password: string,
        status: boolean
)    {
        this.id = id;
        this.nome = nome;
        this.role = role;
        this.departamento = departamento;
        this.usuario = usuario;
        this.password = password;
        this.status = status;

    }
}